### ISSUE TITLE
AssertionFailed when triggering survey

### ISSUE THREAD

**Issue by grmoon** [GoogleCloudPlatform/cloud-code-intellij#2867]
IDE Type: PyCharm
IDE Version: 2020.3
Plugin Version: 20.12.1-202
OS: Mac OS X

```
java.lang.Throwable: Assertion failed
	at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:179)
	at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:188)
	at com.intellij.openapi.wm.impl.WindowWatcher.suggestParentWindow(WindowWatcher.java:238)
	at com.intellij.openapi.wm.impl.WindowManagerImpl.suggestParentWindow(WindowManagerImpl.kt:237)
	at com.google.cloud.tools.intellij.util.ProjectsKt.getCurrentProject(Projects.kt:29)
	at com.google.cloud.tools.intellij.hats.HatsActivationManager.recordActionInBackground$default(HatsActivationManager.kt:67)
	at com.google.cloud.tools.intellij.hats.HatsActivationManager.recordActionInBackground(HatsActivationManager.kt)
	at com.google.cloud.tools.intellij.analytics.ClearcutUsageTrackerService.trackEvent(ClearcutUsageTrackerService.java:361)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.metrics.SkaffoldMetricsKt.sendGenericSkaffoldTrackingEvent(SkaffoldMetrics.kt:151)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.metrics.SkaffoldMetricsKt.sendGenericSkaffoldTrackingEvent$default(SkaffoldMetrics.kt:149)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.metrics.SkaffoldMetricsKt.sendSkaffoldBuildEventMetric(SkaffoldMetrics.kt:186)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.events.SkaffoldEventHandler$addCommonEventListeners$2.onBuildEvent(SkaffoldEventHandler.kt:259)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.events.SkaffoldEventHandler.handleBuildEvent(SkaffoldEventHandler.kt:163)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.events.SkaffoldEventHandler$listenEvents$1.onNext(SkaffoldEventHandler.kt:98)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.events.SkaffoldEventHandler$listenEvents$1.onNext(SkaffoldEventHandler.kt:82)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onMessage(ClientCalls.java:421)
	at io.grpc.ForwardingClientCallListener.onMessage(ForwardingClientCallListener.java:33)
	at io.grpc.ForwardingClientCallListener.onMessage(ForwardingClientCallListener.java:33)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1MessagesAvailable.runInContext(ClientCallImpl.java:519)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by grmoon** [comment]
Kamino cloned this issue to [GoogleCloudPlatform/cloud-code-intellij-internal](https://github.com/GoogleCloudPlatform/cloud-code-intellij-internal/issues/3212)

**Comment by etanshaul** [comment]
reducing priority after investigation: issue is an edge case where the user managed to trigger the survey when the IDE project was closed (maybe closed the project very quickly).

**Comment by ivanporty** [comment]
Fixed in 21.2.1
